codegen-core/src/main/kotlin/io/vrap/rmf/codegen/rendring/CoreCodeGenerator.kt

```diff
@@ -46,7 +44,7 @@ class CoreCodeGenerator @Inject constructor(val dataSink: DataSink
             LOGGER.info("generating files for object types")
             objectTypeGenerators.flatMap { objectTypeRenderer ->
                 allObjectTypes.map { objectTypeRenderer.render(it) }
-            }.map { dataSink.write(it) }
+            }.filter{it.content.isNotEmpty()}. map { dataSink.write(it) } //filter because object type class renderer sometimes doesn't return anything
```

Comment:
I would expect that the render method always returns an non empty string? Why does it make sense to render an empty file?

Reply:
Some entities are marked as **abstact**, like for example Reference.raml. It has "typeId" and "id" fields but no "obj". So we have for example: CategoryReference that extends Reference and contains "obj". What this means is that it makes no sense to generate Reference implementation class, but only the interface which CategoryReference will extend. So, when the class generator gets the Reference entity passed to it, it should not generate anything, therefore it returns an empty string.

Reply:
its just putting that logic later might bite back later, it's like filtering by template content which i don't like.
you can use a file producer instead, then filter the objects to whom you don't want to generate anything.